# SingleCellNotes

## General
* [awesome-single-cell](https://github.com/seandavi/awesome-single-cell): List of software packages (and the people developing these methods) for single-cell data analysis, including RNA-seq, ATAC-seq, etc.
* [scRNA-tools](https://www.scrna-tools.org/): A catalogue of tools for analysing single-cell RNA sequencing data.
* A curated [database](https://docs.google.com/spreadsheets/d/1En7-UV0k0laDiIfjFkdn7dggyR7jIk3WH8QgXaMOZF0/edit#gid=0) of of single-cell transcriptomics studies.
* [Single-cell reading list](https://github.com/gtca/single-cell-reading-list): A curated selection of blog posts and papers on single-cell data analysis. 
* [Single Cell Genomics Day](https://satijalab.org/scgd/): Each year the lag of Rahul Satija organizes a one-day workshop highlighting recent developments in the field.

## Courses
* [Analysis of single cell RNA-seq data](https://scrnaseq-course.cog.sanger.ac.uk/website/index.html): A long-running course from the Cambridge Bioinformatics training unit (Martin Hemberg and others). See also their GitHub [repository](https://github.com/hemberg-lab/scRNA.seq.course). This [course](https://broadinstitute.github.io/2020_scWorkshop/) at the Broad Institute is based on it and offers some interesting extensions (on CITE-Seq for example).
* [MGC/BioSB Course - Single Cell Analysis](https://github.com/LeidenCBC/MGC-BioSB-SingleCellAnalysis2020): This course covers the practicalities of single-cell sample prep and analysis with a particular focus on single-cell RNA-seq libraries. 
* [An introduction to single-cell RNA-seq](https://colab.research.google.com/github/pachterlab/kallistobustools/blob/master/notebooks/Introduction_single_cell_RNA_seq.ipynb#scrollTo=a-eRv9kSISgz):  A Google Colab notebook on the key concepts of single-cell RNA-seq technologies, with a focus on droplet based methods.

## Tutorials
* [Orchestrating Single-Cell Analysis with Bioconductor](http://bioconductor.org/books/release/OSCA/): Very comprehensive on-line book that
teaches you how to make use of cutting-edge Bioconductor tools to process, analyze, visualize, and explore scRNA-seq data. The companion paper can be found [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7358058/).
* [Current best practices in single‐cell RNA‐seq analysis: a tutorial](https://www.embopress.org/doi/full/10.15252/msb.20188746): A highly readable tutorial by Malte Luecken
 and Fabian Theis. There is a companion GitHub [repository](https://github.com/theislab/single-cell-tutorial) with the scripts
* [Seurat vignettes](https://satijalab.org/seurat/vignettes.html): Seurat remains the most comprehensive R toolkit for single genomics analysis accompanied by a large collection of vignettes. 

## Trajectory inference

* [single-cell-pseudotime](https://github.com/agitter/single-cell-pseudotime): Overview of single-cell RNA-seq pseudotime estimation algorithms.
* [dynmethods](https://github.com/dynverse/dynmethods): A collection of 55 trajectory inference methods. To run any of these methods, interpret the results and visualise the trajectory, see the [dyno](https://github.com/dynverse/dyno) package.

## Spatial transcriptomics

## Gene regulatory network inference
